docs/machine-learning/tutorials/movie-recommmendation.md

```diff
@@ -0,0 +1,474 @@
+There are three major concepts in ML.NET: [Data](../basic-concepts-model-training-in-mldotnet.md#data), [Transformers](../basic-concepts-model-training-in-mldotnet.md#transformer), and [Estimators](../basic-concepts-model-training-in-mldotnet.md#estimator).
+
+Machine learning training algorithms require data in a certain format. `Transformers` are used to transform tabular data to a compatible format.
+
+![transformer image](./media/movie-recommendation/transformer.png)
+
+You create `Transformers` in ML.NET by creating `Estimators`. `Estimators` take in data and return `Transformers`.
+
+![estimator image](./media/movie-recommendation/estimator.png)
+
+The recommendation training algorithm you will use for training your model is an example of an `Estimator`.
+
+Build an `Estimator` with the following steps:
+
+Create the `BuildAndTrainModel()` method, just after the `LoadData()` method, using the following code:
+```csharp
+public static ITransformer BuildAndTrainModel(MLContext mlContext, IDataView trainingDataView)
+{
+
+
+}
+```
+> [!NOTE]
+> This method will give you an error until you add a return statement in the following steps.
+
+Define the data transformations by adding the following code to `BuildAndTrainModel()`:
+   
+[!code-csharp[DataTransformations](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#DataTransformations "Define data transformations")]
+
+Since `userId` and `movieId` represent users and movie titles, not real values, you use the [MapValueToKey()](xref:Microsoft.ML.ConversionsExtensionsCatalog.MapValueToKey%2A) method to transform each `userId` and each `movieId` into a numeric key type `Feature` column (a format accepted by recommendation algorithms) and add them as new dataset columns:
+
+| userId | movieId | Label | userIdEncoded | movieIdEncoded |
+| ------------- |:-------------:| -----:|-----:|-----:|
+| 1 | 1 | 4 | userKey1 | movieKey1 |
+| 1 | 3 | 4 | userKey1 | movieKey2 |
+| 1 | 6 | 4 | userKey1 | movieKey3 |
+
+
+Choose the machine learning algorithm and append it to the data transformation definitions by adding the following as the next line of code in `BuildAndTrainModel()`:
+
+[!code-csharp[AddAlgorithm](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#AddAlgorithm "Add the training algorithm with options")]
+
+The [MatrixFactorizationTrainer](xref:Microsoft.ML.RecommendationCatalog.RecommendationTrainers.MatrixFactorization%28Microsoft.ML.Trainers.MatrixFactorizationTrainer.Options%29) as is your recommendation training algorithm.  [Matrix Factorization](https://en.wikipedia.org/wiki/Matrix_factorization_(recommender_systems)) is a common approach to recommendation when you have data on how users have rated products in the past, which is the case for the datasets in this tutorial. There are other recommendation algorithms for when you have different data available (see the [Recommended Resources](#recommended-resources) section below to learn more). 
```

Comment:
Also, need to update [Recommended Resources](#recommended-resources) --> [Resources](#resources)